### ISSUE TITLE
Provide a compiles landclim version in the package?

### ISSUE THREAD

**Issue by KDolos** [KIT-IfGG/LandClimTools#4]
-> for Ubuntu, Windows and Mac. Would be nice, but is it allowed?

**Comment by floscher** [comment]
Absolutely. LandClim is free software under the GPLv3 license and thus can be distributed, modified and executed in binary or source form. Only requirement is, that you also publish the license text along with it and modified versions of the software have to be under the same license if they are distributed.

But it would be better (e.g. more portable) if the *source code* would be distributed with the package. On installation of the package it would then be compiled.
Because of this, a package containing pre-compiled binaries is not allowed in CRAN:
> A source package if possible should not contain binary executable files: they are not portable, and a security risk if they are of the appropriate architecture. `R CMD check` will warn about them unless they are listed (one filepath per line) in a file `BinaryFiles` at the top level of the package. Note that CRAN will not accept submissions containing binary files even if they are listed.
>
> (see https://cran.r-project.org/doc/manuals/R-exts.html#Package-structure)

Both options (including binary or source code)would be entirely allowed and possible.

**Comment by aushaff** [comment]
Is any work currently being done on this?

**Comment by KDolos** [comment]
Unfortunately not.

**Comment by KDolos** [comment]
LandClim can be accessed freely from ETH now. A compiled version for WIn is available. For Other platforms the source code needs to be downloaded and compiled, as far as I understand.
https://uwis-server102.ethz.ch/openaccess/

**Comment by floscher** [comment]
AFAICS the situation for downloading source/binaries hasn't changed, you'll still have to create an account in order to download the software.

**Comment by KDolos** [comment]
Yes, that's true, but they will give you the executable, takes just a few hours or days, if the responsible person is on holiday, I assume. The code is not generally open for a couple of reasons; publications in progress, maintenance (who will answer all the questions...?!) an others. Mabe @TimothyThrippleton  and @aushaff could tell more about this issue?

**Comment by caspargross** [comment]
Hey everyone!
I am glad to see that a lot of stuff is happening around this R package. Today I wanted to look at the current state of LandCLim and compiled the current version from Source (V1.6 rev 726).
Maybe some of the observations I did along the way could be helpful if this package will ever include the source and compile directly.

### Dependencies
1) Gnu C Compiler (g++) Should already be installed on Linux
2) Boost libraries. Install on ubuntu with
   `sudo apt-get install libboost-all-dev`
3) This makefile:
   <script src="https://gist.github.com/caspargross/b5b441f8c782ab59a59a19cab866e27a.js"></script>

### Compile
1) Download ZIP with source code from ETH and extract
2) Download makefile and place in LandClim folder
2) Make sure that your boost libraries are installed to `/usr/lib/x86_64` by checking:
   `locate libboost`
   If they are somewhere else you need to adapt the path in line 5 of the makefile
3) Fix all the bugs in the sourcecode (see below)

### Bugs (only those that prevent compilation)

1) In file `src/pasture.hpp` line 45:
  `boolean is_pasture`
  to
  `bool is_pasture`

2) In `src/climate.cpp` delete or comment out on line 26:
  `#include <windows.h>`

3) In `src/landClimGlobals.hpp` change on line 26:
  `#include <istream>`
  to
  `#include <ifstream>`

4)) In `src/landClimGlovals.cpp` change on line 761:
  `if (source != NULL )`
  to
  `if (source.is_open())`

5) In `src/cohort.cpp` change on line 175 *and* 250:
  `boolean justcut`
  to
  `bool justcut`

6) In src/cohort.cpp change all Uppercase TRUE and FALSE to lowercase true and false


7) Deactivate lots of LandClim elements (grass, undergrow, litter, fire) that throw errors during compilation. This leads to the loss of these functions.
Comment out the following rows in src/landclim.cpp
`cohorts.putLitter.connect(bind(&Fuel::onLitter, &fuel, _1, _2));
cohorts.putGrassHarvest.connect(bind(&GrassObserver::onHarvest, &grassObserver, _1, _2));
cohorts.putGrassGrowth.connect(bind(&GrassObserver::onYearEnd, &grassObserver, _1, _2));

cohorts.putUnderstoreyGrowth.connect(bind(&GrassObserver::UnderstoreyYearlyBiomass, &grassObserver, _1, _2));
fuel.putNewFuelEvent.connect(bind(&FuelObserver::onNewEvent, &fuelObserver));
fuel.putFuelInfo.connect(bind(&FuelObserver::onEventInfo, &fuelObserver, _1, _2));
`

If someone wants the binaries or patched files, send me a message!
Cheers, Caspar

**Comment by KDolos** [comment]
I think, in the new version of LandClim this is fixed, @TimothyThrippleton an @dtmichel will know. Or Tim can use these infos in the Tutorial.

Nice to hear from you again :-)!

**Comment by dtmichel** [comment]
Yes, the errors that break the build on Linux and MacOS, as mentioned by @caspargross, are fixed in version 1.7. Also a make file is included. 
The papers for version 1.7 have not been yet been published, which is why we didn't put 1.7 on OpenAccess yet.


The point no. 7 from the list above doesn't ring a bell, I never experienced this and never had to deactivate code in LandClim.

**Comment by dtmichel** [comment]
LandClim 1.7 has been officially released:
https://uwis-server102.ethz.ch/openaccess/packages/view/239
